src/api/alchemy-websocket-provider.ts

```diff
@@ -687,22 +691,16 @@ export class AlchemyWebSocketProvider
   /** @internal */
   private customStartEvent(event: EthersEvent): void {
     if (event.type === 'alchemy') {
-      const { address } = event;
-      if (!!address) {
-        void this._subscribe(
-          event.tag,
-          ['alchemy_filteredNewFullPendingTransactions', { address }],
-          this.emitProcessFn(event),
-          event
-        );
-      } else {
-        void this._subscribe(
-          event.tag,
-          ['alchemy_newFullPendingTransactions'],
-          this.emitProcessFn(event),
-          event
-        );
-      }
+      const { fromAddress, toAddress, hashesOnly } = event;
+      void this._subscribe(
+        event.tag,
+        [
+          'alchemy_pendingTransactions',
+          stripUndefined({ fromAddress, toAddress, hashesOnly })
```

Comment:
Why's the `stripUndefined` needed? Seems like this would only matter if you plan to iterate over the keys, which I don't see happening anywhere.

Reply:
thanks for prompting me to dig deeper! I originally had tests fail since undefined fields were present, but I realize that the test was stubbing out `send()`, rather than the underlying websocket server. Removed the method entirely.